dc-cudami-editor/src/components/PagedIdentifiableList.jsx

```diff
@@ -0,0 +1,266 @@
+    const {apiContextPath, mockApi, parentType, parentUuid, type} = this.props
+    const successful = await addAttachedIdentifiables(
+      apiContextPath,
+      mockApi,
+      identifiables,
+      parentType,
+      parentUuid,
+      type
+    )
+    return successful
+  }
+
+  handleAdd = async (identifiablesToAdd) => {
+    const {identifiables, pageNumber} = this.state
+    const uniqueIdentifiables = this.makeUniqueList(identifiablesToAdd)
+    const successful = await this.addIdentifiables(uniqueIdentifiables)
+    if (!successful) {
+      return console.error('an error occured')
+    }
+    if (identifiables.length === this.pageSize) {
+      return this.updatePage({selected: pageNumber + 1})
+    }
+    this.updatePage({selected: pageNumber})
+  }
+
+  handleRemove = async () => {
+    const {identifiables, pageNumber, removeIndex} = this.state
+    const uuid = identifiables[removeIndex].uuid
+    const successful = await this.removeIdentifiable(uuid)
+    if (!successful) {
+      return console.error('an error occured')
+    }
+    if (pageNumber > 0 && identifiables.length === 1) {
+      return this.updatePage({selected: pageNumber - 1})
+    }
+    this.updatePage({selected: pageNumber})
+  }
+
+  loadIdentifiables = async (pageNumber = 0) => {
+    const {apiContextPath, mockApi, parentType, parentUuid, type} = this.props
+    const identifiables = await loadAttachedIdentifiables(
+      apiContextPath,
+      mockApi,
+      parentType,
+      parentUuid,
+      type,
+      pageNumber,
+      this.pageSize
+    )
+    return identifiables
+  }
+
+  makeUniqueList = (identifiables) => {
+    const existingUuids = this.state.identifiables.map(({uuid}) => uuid)
+    const uniqueIdentifiables = uniqBy(identifiables, 'uuid')
+    return uniqueIdentifiables.filter(({uuid}) => {
+      return !existingUuids.includes(uuid)
+    })
+  }
+
+  removeIdentifiable = async (uuid) => {
+    const {apiContextPath, mockApi, parentType, parentUuid, type} = this.props
+    const successful = await removeAttachedIdentifiable(
+      apiContextPath,
+      mockApi,
+      parentType,
+      parentUuid,
+      type,
+      uuid
+    )
+    return successful
+  }
+
+  toggleModal = (name) => {
+    this.setState({
+      modalsOpen: {
+        ...this.state.modalsOpen,
+        [name]: !this.state.modalsOpen[name],
+      },
+    })
+  }
+
+  updatePage = async ({selected}) => {
+    const {content, pageSize, totalElements} = await this.loadIdentifiables(
+      selected
+    )
+    this.setState({
+      identifiables: content,
+      numberOfPages: Math.ceil(totalElements / pageSize),
+      pageNumber: selected,
+    })
+  }
+
+  render() {
+    const {
+      apiContextPath,
+      debug,
+      mockApi,
+      parentType,
+      t,
+      type,
+      uiLocale,
+    } = this.props
+    return (
+      <ApiContext.Provider value={{apiContextPath, mockApi}}>
+        <Row>
+          <Col>
+            <h2>{t(`${type}s`)}</h2>
```

Comment:
Hast du zufällig ein Beispiel, was `${type}s` sein kann?

Reply:
Die Variable `type` wäre in unserem Anwendungsfall `digitalObject`.